packages/app-page-builder/src/render/plugins/elements/menu/components/MenuItems/findObject.ts

```diff
@@ -0,0 +1,26 @@
+import { findIndex, each } from "lodash";
+
+/**
+ * Recursively search for an object with given ID in the given source array.
+ */
+const findObject = (source: Array<any>, id: string): any => {
```

Comment:
This is ok, but let's try to avoid lodash when possible :)
Here is how it would look with raw JS:
```
type FindObjectResult<T> = {
  source: T[],
  index: number,
  item: T,
};
const findObject = <T extends any>(source: T[], id: string): FindObjectResult<any> | null => {
  if (source.length === 0) {
    return null;
  }
  const index = source.findIndex(item => item.id === id);
  if (index >= 0) {
    return {source, index, item: source[index]};
  }
  return source.reduce((acc: FindObjectResult<T> | null, item: T) => {
    if (!item.children) {
      return acc;
    }
    return findObject(item.children, id);
  }, null);
};
```

Didn't run the fn but it should work :)